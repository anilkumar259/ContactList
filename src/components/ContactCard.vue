<template>
    <div class="contact-card" v-for="contact in contactMembersList" :key="contact.login.uuid">
        <div class="contact-photo">
              <img :src="contact.picture.large" :alt="`${contact.name.first} ${contact.name.last}`" />
        </div>
        <div class="contact-info">
            <div class="contact-name">{{ contact.name.first }} {{ contact.name.last }}</div>
            <div class="contact-details">
                <div class="contact-address">
                    {{contact.location.street.number}} {{contact.location.street.name}} <br>
                     {{ contact.location.postcode }} {{contact.location.city}} {{contact.location.state}}<br>
                    {{contact.location.country}} 
                </div>
                <button class="call-button">Call {{ contact.cell }}</button>
            </div>
        </div>
    </div>
</template>
<script setup>
import { defineProps } from 'vue';
const props = defineProps({
    contactMembersList: {
        type: Array,
        required: true
    }
});
</script>