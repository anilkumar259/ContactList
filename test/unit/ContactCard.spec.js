import { mount } from '@vue/test-utils'
import ContactCard from '@/components/ContactCard.vue'

describe('ContactCard.vue', () => {
  it('start checking with sample data', () => {
    const validateContacts = [
      {
        login: { uuid: "56ae5552-a4f2-4dcb-a88a-86c9cd7098ea" },
        name: { first: "anil", last: "kumar" },
        email: "test@gmail.com",
        picture: { large: "sample.jpg" },
        location: {
          street: { number: 6549, name: "Madhapur" },
          city: "hyd",
          state: "telegana",
          country: "India",
          postcode: 500081
        },
        cell: "(123) 123-1234",
      },

    ]
    const wrapper = mount(ContactCard, {
      props: { contactMembersList: validateContacts }
    })

    expect(wrapper.text()).toContain('anil kumar')
  })
})
