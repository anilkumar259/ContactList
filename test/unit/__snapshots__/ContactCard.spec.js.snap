// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ContactCard.vue start checking with sample data 1`] = `
<div class="contact-card">
  <div class="contact-photo"><img src="sample.jpg" alt="anil kumar"></div>
  <div class="contact-info">
    <div class="contact-name">anil kumar</div>
    <div class="contact-details">
      <div class="contact-address">6549 Madhapur <br> 500081 hyd telegana<br> India</div><button class="call-button">Call (123) 123-1234</button>
    </div>
  </div>
</div>
`;
